From bcd5790011ca13f23df34599b7598688f09bf571 Mon Sep 17 00:00:00 2001
From: Hariprasad Kelam <hkelam@marvell.com>
Date: Mon, 21 Sep 2020 15:29:20 +0530
Subject: [PATCH 12/22] octeontx2-af: Pass default values incase user did not
 specify

Current implementation is such that if user did not specify
parameters like speed/duplex/autoneg its fills AUTONEG_UNKNOWN,
SPEED_UNKNOWN,DUPLEX_UNKNOWN. In this case pass default values
defined in othercase pass user specified values.

Change-Id: I6eabc051ed4a80e9dc597c25d860b456ad92f2bb
Signed-off-by: Hariprasad Kelam <hkelam@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/38065
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Integrated-by: Abhishek Paliwal <paliwal.abhishek@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/cgx.c  | 9 ++++++---
 drivers/net/ethernet/marvell/octeontx2/af/mbox.h | 1 +
 2 files changed, 7 insertions(+), 3 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/cgx.c b/drivers/net/ethernet/marvell/octeontx2/af/cgx.c
index 93c642d8e50c..a579ae744b37 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/cgx.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/cgx.c
@@ -953,10 +953,13 @@ static void set_mod_args(struct cgx_set_link_mode_args *args,
 			 u32 speed, u8 duplex, u8 autoneg, u64 mode)
 {
 	/* firmware requires this value in the reverse format */
-	args->duplex = duplex;
-	args->speed = speed;
+	if (args->duplex == DUPLEX_UNKNOWN)
+		args->duplex = duplex;
+	if (args->speed == SPEED_UNKNOWN)
+		args->speed = speed;
+	if (args->an == AUTONEG_UNKNOWN)
+		args->an = autoneg;
 	args->mode = mode;
-	args->an = autoneg;
 	args->ports = 0;
 }
 
diff --git a/drivers/net/ethernet/marvell/octeontx2/af/mbox.h b/drivers/net/ethernet/marvell/octeontx2/af/mbox.h
index e43765150d25..2778a13c35ef 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/mbox.h
+++ b/drivers/net/ethernet/marvell/octeontx2/af/mbox.h
@@ -616,6 +616,7 @@ struct cgx_set_link_mode_args {
 };
 
 struct cgx_set_link_mode_req {
+#define AUTONEG_UNKNOWN		0xff
 	struct mbox_msghdr hdr;
 	struct cgx_set_link_mode_args args;
 };
-- 
2.17.1

